### PR TITLE
Chore: spelling - pkg - excluding api changes

### PR DESCRIPTION
**Pull request by jsoref**
**What this PR does / why we need it**:

This fixes a number of spelling errors.

**Special notes for your reviewer**:

Comments on file changes can be found in #24412

### CODE REVIEW

File: pkg/services/alerting/notifiers/base.go [hunk 1]
```diff
@@ -10,15 +10,15 @@ import (
 )
 
 const (
-	triggMetrString = "Triggered metrics:\n\n"
+	triggeredMetricsString = "Triggered metrics:\n\n"
```
Comment (aknuds1) [inline pkg/services/alerting/notifiers/base.go:13]: Please undo this change, as it's completely subjective.

File: pkg/services/alerting/notifiers/kafka.go [hunk 2]
```diff
@@ -64,7 +64,7 @@ type KafkaNotifier struct {
 func (kn *KafkaNotifier) Notify(evalContext *alerting.EvalContext) error {
 	state := evalContext.Rule.State
 
-	customData := triggMetrString
+	customData := triggeredMetricsString
```
Comment (aknuds1) [inline pkg/services/alerting/notifiers/kafka.go:67]: Please undo this change.

File: pkg/services/alerting/notifiers/opsgenie.go [hunk 3]
```diff
@@ -110,7 +110,7 @@ func (on *OpsGenieNotifier) createAlert(evalContext *alerting.EvalContext) error
 		return err
 	}
 
-	customData := triggMetrString
+	customData := triggeredMetricsString
```
Comment (aknuds1) [inline pkg/services/alerting/notifiers/opsgenie.go:113]: Please undo this change.

File: pkg/services/sqlstore/migrations/dashboard_mig.go [hunk 4]
```diff
@@ -42,7 +42,7 @@ func addDashboardMigration(mg *Migrator) {
 	}
 
 	mg.AddMigration("create dashboard_tag table", NewAddTableMigration(dashboardTagV1))
-	mg.AddMigration("add unique index dashboard_tag.dasboard_id_term", NewAddIndexMigration(dashboardTagV1, dashboardTagV1.Indices[0]))
+	mg.AddMigration("add unique index dashboard_tag.dashboard_id_term", NewAddIndexMigration(dashboardTagV1, dashboardTagV1.Indices[0]))
```
Comment (aknuds1) [inline pkg/services/sqlstore/migrations/dashboard_mig.go:45]: @marefr Might this break things?

File: pkg/tsdb/testdatasource/scenarios.go [hunk 5]
```diff
@@ -47,7 +47,7 @@ func init() {
 			factor := 2
 			for i := 0; i < 10; i++ {
 				timeWalkerMs := context.TimeRange.GetFromAsMsEpoch()
-				serie := &tsdb.TimeSeries{Name: strconv.Itoa(start)}
+				timeSeries := &tsdb.TimeSeries{Name: strconv.Itoa(start)}
```
Comment (aknuds1) [inline pkg/tsdb/testdatasource/scenarios.go:50]:
```suggestion
				series := &tsdb.TimeSeries{Name: strconv.Itoa(start)}
```

File: pkg/tsdb/testdatasource/scenarios.go [hunk 6]
```diff
@@ -77,7 +77,7 @@ func init() {
 			var series []*tsdb.TimeSeries
 			for i := 0; i < 10; i++ {
 				timeWalkerMs := context.TimeRange.GetFromAsMsEpoch()
-				serie := &tsdb.TimeSeries{Name: strconv.Itoa(i * 10)}
+				timeSeries := &tsdb.TimeSeries{Name: strconv.Itoa(i * 10)}
```
Comment (aknuds1) [inline pkg/tsdb/testdatasource/scenarios.go:80]:
```suggestion
				series := &tsdb.TimeSeries{Name: strconv.Itoa(i * 10)}
```

**Review by aknuds1 — CHANGES_REQUESTED** [review]

File: pkg/services/alerting/eval_handler_test.go [hunk 7]
```diff
@@ -153,7 +153,7 @@ func TestAlertingEvaluationHandler(t *testing.T) {
 			So(context.ConditionEvals, ShouldEqual, "[[false OR false] OR false] = false")
 		})
 
-		Convey("Should retuasdfrn no data if one condition has nodata", func() {
+		Convey("Should return no data if one condition has nodata using OR operator", func() {
```
[the remark this answers is not in the document]
Comment (jsoref) [reply pkg/services/alerting/eval_handler_test.go:156]: Because the same precise convey string is used in the next convey call, so I needed to do something, otherwise the CI failed.

I don't care what the words were, but I needed something.

The code only accidentally worked due to that keyboard inserted damage.

File: pkg/tsdb/testdatasource/scenarios.go (hunk 5 above)
Comment (jsoref) [reply pkg/tsdb/testdatasource/scenarios.go:50]: I can't do this, there's a series variable on the next line. I hit that in CI as well.

**Comment by jsoref** [comment]
@sakjur ?

File: pkg/services/alerting/eval_handler_test.go (hunk 7 above)
[the remark this answers is not in the document]
Comment (jsoref) [reply pkg/services/alerting/eval_handler_test.go:156]: https://github.com/grafana/grafana/commit/bf999a4f826aab98c3ecfea94bbd16cfb4d1abc7#diff-34cb3b27b2b7c3d3f53d31e812bbd169R158-R171

@bergquist

**Comment by aknuds1** [comment]
Please merge with master, as we merged a subset of your changes in another PR (so we could isolate changes we were certain would not break anything).

File: pkg/services/sqlstore/migrations/dashboard_mig.go (hunk 4 above)
Comment (marefr) [reply pkg/services/sqlstore/migrations/dashboard_mig.go:45]: Yes

Comment (jsoref) [reply pkg/services/sqlstore/migrations/dashboard_mig.go]: Dropped. I can add an exclusion for it...

**Review by aknuds1 — CHANGES_REQUESTED** [review]
See suggestions, I don't see any point in changing test strings.

**Comment by aknuds1** [comment]
Please merge with latest master, hopefully it fixes the CI failure.

**Comment by jsoref** [comment]
Typos in tests are problematic too, and adding exemptions makes it harder to prevent/catch typos in core code. But, my goal is to improve where I can. Dropped.

File: pkg/services/alerting/notifiers/base.go [hunk 8]
```diff
@@ -18,7 +18,7 @@ type NotifierBase struct {
 	Name                  string
 	Type                  string
 	UID                   string
-	IsDeault              bool
+	IsDefault             bool
```
Comment (aknuds1) [inline pkg/services/alerting/notifiers/base.go:21]: @marefr @bergquist Do you know if it's safe to rename a field on `NotifierBase`? I'm just wondering if it could affect external entities, e.g. through notifiers being sent.

Comment (marefr) [reply pkg/services/alerting/notifiers/base.go:21]: Based on search of usage should be safe

Comment (aknuds1) [reply pkg/services/alerting/notifiers/base.go:21]: Great, thanks for clarifying that @marefr.

**Review by aknuds1 — APPROVED** [review]
LGTM

**Comment by aknuds1** [comment]
Thanks for contributing to Grafana!
